.golangci.yaml

```diff
@@ -92,6 +92,16 @@ linters:
     #- wsl
 
 linters-settings:
+  depguard:
+    rules:
+      main:
+        list-mode: original
+        allow:
+          - $gostd
+          - github.com/NVIDIA
+          - github.com/go-logr/logr
+          - k8s.io
+          - sigs.k8s.io
```

Comment:
Question -- why was this change made as part of this PR?

Reply:
`depguard` was one of the linters enabled on this repo and it started failing with the latest golangci-lint versions. Adding these depguard settings allow lists all of the dependencies imported by k8s-operator-libs